telegram/base.py

```diff
@@ -80,3 +80,25 @@ def to_dict(self):
                     data[key] = value
 
         return data
+
+    def _get_id(self):
+        for x in ['id', 'file_id', 'message_id', 'result_id', 'update_id']:
+            if hasattr(self, x):
+                return self.__getattribute__(x)
+        raise NotImplementedError
+
+    def __eq__(self, other):
+        if isinstance(other, self.__class__):
+            try:
+                return self._get_id() == other._get_id()
+            except NotImplementedError:
+                pass  # return NotImplemented
+        return NotImplemented
+
+    def __hash__(self):
+        try:
+            return hash((
+                self.__class__,
+                self._get_id(),))
+        except NotImplementedError:
```

Comment:
I don't think catching `NotIMplementedError` here is needed. If we get such exception, it's probably a bug which we'll mask,